### DISCUSSION TITLE
Why does it say ERROR? What am I doing wrong? | Writing my own parser

### DISCUSSION THREAD

**Discussion by Samvid95** [tree-sitter/tree-sitter#3105] [Q&A] [answered]
Hi there, I am new to using tree-sitter and writing parser and appreciate some help on what I am doing wrong. 

```
#ifdef ABC123
    FORM S123.F123;
#endif
```

Now for this section here's my grammar file: 
```
rules: {
        source_file: $ => seq(
          repeat($._statement)
        ),
    
        _statement: $ => choice(
            $.conditional_block,
            $.form_construct,
          // Add more statement types here as necessary.
        ),
    
        form_construct: $ => seq(
            'FORM',
            field('formset_id', $.identifier),
            '.',
            field('form_id', $.identifier),
            ';'
        ),
        conditional_block: $ => choice(
            $.ifdef_block,
          ),
      
        ifdef_block: $ => seq(
            field('if_directive_start', /#ifdef/),
            field('condition', $.identifier),
            repeat($._statement),
            field('if_directive_end', /#endif/),
        ),
      
        identifier: $ => /[A-Za-z_][A-Za-z0-9_]*/,
        } 
```

Here's my output: 
```
(source_file [0, 0] - [2, 6]
  (ERROR [0, 0] - [0, 13]
    (ERROR [0, 7] - [0, 13]))
  (form_construct [1, 4] - [1, 27]
    formset_id: (identifier [1, 9] - [1, 20])
    form_id: (identifier [1, 21] - [1, 26]))
  (ERROR [2, 0] - [2, 6]))
```
I am not sure why it just comes out as ERROR? The form_construct works but ifdef block is not working. Any help is appreciated.

**Comment by WillLillis** [comment, marked as answer]
I didn't get any errors when trying to parse the test the sample input with your grammar. Here's a screenshot:

![image](https://github.com/tree-sitter/tree-sitter/assets/87077023/a693aa01-18d4-48c9-bc4e-a29e46ae9d4f)

Is it possible you didn't use the `tree-sitter generate` command to regenerate the grammar after some changes were made to your `grammar.js`?

**Comment by Samvid95** [comment]
Well I guess you are right. I may have just not did tree-sitter generate. Thank you so much for running the code.
